src/cunumeric/fft/fft.cu

```diff
@@ -0,0 +1,355 @@
+    size_t num_elements_in  = 1;
+    size_t num_elements_out = 1;
+    for(int i = 0; i < DIM; ++i) {
+      n[i]          = (direction == fftDirection::FFT_FORWARD) ? fft_size_in[i] : fft_size_out[i];
+      inembed[i]    = fft_size_in[i];
+      onembed[i]    = fft_size_out[i];
+      num_elements_in  *= fft_size_in[i];
+      num_elements_out *= fft_size_out[i];
+    }
+
+    // Copy input to temporary buffer to perform FFTs one by one
+    DeferredBuffer<INPUT_TYPE, DIM> input_buffer(Rect<DIM>(zero, fft_size_in - one),
+                                                 Memory::GPU_FB_MEM,
+                                                 nullptr /*initial*/,
+                                                 128 /*alignment*/);
+    CHECK_CUDA(cudaMemcpyAsync(input_buffer.ptr(zero), in.ptr(zero), num_elements_in*sizeof(INPUT_TYPE), cudaMemcpyDefault, stream));
+
+    DeferredBuffer<OUTPUT_TYPE, DIM> output_buffer(Rect<DIM>(zero, fft_size_out - one),
+                                                  Memory::GPU_FB_MEM,
+                                                  nullptr /*initial*/,
+                                                  128 /*alignment*/);
+    // Create the plan
+    cufftHandle plan;
+    CHECK_CUFFT(cufftCreate(&plan));
+    CHECK_CUFFT(cufftSetAutoAllocation(plan, 0 /*we'll do the allocation*/));
+    CHECK_CUFFT(cufftSetStream(plan, stream));
+
+    // Operate over the R2C or C2R axes, which is the first one
+    auto axis = axes.front();
+
+    // Batched 1D dimension
+    dim_t size_1d = n[axis];
+    // TODO: batch only correct for DIM <= 3. Fix for N-DIM case
+    dim_t batches = (direction == fftDirection::FFT_FORWARD) ? num_elements_in : num_elements_out;
+                  batches = (DIM == 3 && axis == 1) ? n[2] : batches / n[axis];
+    dim_t istride = 1;
+    dim_t ostride = 1;
+    for(int i = axis+1; i < DIM; ++i) {
+      istride *= fft_size_in[i];
+      ostride *= fft_size_out[i];
+    }
+    dim_t idist = (axis == DIM-1) ? fft_size_in[axis]  : 1;
+    dim_t odist = (axis == DIM-1) ? fft_size_out[axis] : 1;
+
+    // Create the plan and allocate a temporary buffer for it if it needs one
+    CHECK_CUFFT(cufftMakePlanMany64(plan, 1, &size_1d, inembed, istride, idist, onembed, ostride, odist, (cufftType)type, batches, &workarea_size));
+
+    DeferredBuffer<uint8_t, 1> workarea_buffer;
+    if(workarea_size > 0) {
+      const Point<1> zero1d(0);
+      workarea_buffer =
+        DeferredBuffer<uint8_t, 1>(Rect<1>(zero1d, Point<1>(workarea_size - 1)),
+                                   Memory::GPU_FB_MEM,
+                                   nullptr /*initial*/,
+                                   128 /*alignment*/);
+      void* workarea = workarea_buffer.ptr(zero1d);
+      CHECK_CUFFT(cufftSetWorkArea(plan, workarea));
+    }
+
+    cufft_axes_plan<DIM, OUTPUT_TYPE, INPUT_TYPE>::execute(plan, output_buffer, input_buffer, out_rect, in_rect, axis, direction);
```

Comment:
Same comment as previously. Since plan creation / plan execution are tightly coupled, I would put them both in the same place. But up to you.